### PR TITLE
overhaul of homebrew updating script

### PR DESCRIPTION
**Pull request by gdams**
Adds a template to make cask changes in a much simpler way.

### CODE REVIEW

File: auto_updater.sh [hunk 1]
```diff
@@ -1,5 +1,16 @@
 #!/bin/bash set -eu
 
+PUSH=
```
Comment (karianna) [inline auto_updater.sh]: not defaulting to false as opposed to undefined?

File: auto_updater.sh [hunk 2]
```diff
@@ -22,6 +33,13 @@ function check_hub {
 }
 
 function update_casks {
+  cd Casks
```
Comment (karianna) [inline auto_updater.sh]: what if that doesn't exist?

File: auto_updater.sh (hunk 1 above)
Comment (karianna) [reply auto_updater.sh]: comment what that is

File: auto_updater.sh (hunk 2 above)
Comment (gdams) [reply auto_updater.sh]: It always does

File: auto_updater.sh (hunk 1 above)
Comment (gdams) [reply auto_updater.sh]: https://github.com/AdoptOpenJDK/homebrew-openjdk/pull/243/files#diff-74755599760817973315cd18477be295R8

**Pull request merged**